File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int l=0,r=0;
        int n=s.size();
        int maxi=0;
        unordered_set<int> st;
        while(r<n)
        {
            while(st.count(s[r]))
            {
                st.erase(s[l]);
                l++;
            }
            st.insert(s[r]);
            maxi=max(maxi,r-l+1);
            r++;
        }
        return maxi;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> hs=new HashSet<>();
        int n=s.length();
        int l=0,r=0;
        int maxi=0;
        while(r<n)
        {
            while(r<n && !hs.contains(s.charAt(r)))
            {
                hs.add(s.charAt(r));
                r++;
            }
            maxi=Math.max(maxi,r-l);
            while(r<n && l<r && hs.contains(s.charAt(r)))
            {
                hs.remove(s.charAt(l));
                l++;
            }
        }
        return maxi;
    }
}